react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -238,7 +244,7 @@ const getThreeDNodesFromAsset = async (
 };
 
 const applyNodeStyles = (
-  treeNodes: NodeAndRange[],
+  treeNodes: Array<NodeAndRange | undefined>,
```

Comment:
This shouldn't have to include `undefined` - in the `.flat().filter(...)` above, you can use the `filterUndefined` utility function as the filter predicate. It will ensure that the type of the resulting list does not include undefined.

Reply:
Ah ok cool. Changed here:

https://github.com/cognitedata/reveal/pull/4243/commits/a2c40bb426411b21222394823eedb878b665070e